modules/transcription-service-ibm-watson-impl/src/main/java/org/opencastproject/transcription/ibmwatson/IBMWatsonTranscriptionService.java

```diff
@@ -740,12 +774,10 @@ public MediaPackageElement getGeneratedTranscription(String mpId, String jobId)
   }
 
   protected CloseableHttpClient makeHttpClient() {
-    CredentialsProvider credentialsProvider = new BasicCredentialsProvider();
-    credentialsProvider.setCredentials(AuthScope.ANY, new UsernamePasswordCredentials(user, psw));
     RequestConfig reqConfig = RequestConfig.custom().setConnectTimeout(CONNECTION_TIMEOUT)
             .setSocketTimeout(SOCKET_TIMEOUT).setConnectionRequestTimeout(CONNECTION_TIMEOUT).build();
-    return HttpClients.custom().setDefaultCredentialsProvider(credentialsProvider).setDefaultRequestConfig(reqConfig)
-            .build();
+    return HttpClients.custom().setDefaultRequestConfig(reqConfig)
+            .setRetryHandler(new DefaultHttpRequestRetryHandler(3, true)).build();
```

Comment:
Is 3 defined in their spec?  It seems like a reasonable number of attempts, but I'm not a fan of hardcoded values.

Reply:
No, it isn't. I didn't realize this fix wasn't already there... I looked at our ticket history and this was a fix we made more than a year ago for dealing with stale connections in the http client library (we would get a "failed to respond" error, but a re-try of the operation a few minutes later would just work). 3 was an arbitrary number and we didn't see those errors anymore. Again, not reproducible :(
I can make it a constant in the Java class or it could be a configuration option in the service, which I honestly think is overkill. It's always good to retry when using an external service and 3 also seems a good number to me:)

Reply:
Ok. I'm not a huge fan but the choice seems reasonable here. Chances are pretty good that if 3 attempts fail then N will fail too - the service is probably having some downtime. Adding user configurability to this isn't gaining anything but code complexity :) 

Let's just leave this as is.